### PR TITLE
[Keyboard] Add cosmo65

### PR DESCRIPTION
**Pull request by BentoBoxStudio**
## Description

Add cosmo65.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cosmo65/config.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#define RGBLIGHT_ANIMATIONS
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 14
+
+/* keyboard matrix assignments */
+#define MATRIX_ROW_PINS { D5, D0, F0, F5, F4 }
+#define MATRIX_COL_PINS { E6, D4, D6, D7, B4, B5, B6, C6, C7, D1, D2, D3, F1, F6 }
+#define UNUSED_PINS {B0, B7}
```
Comment (waffle87) [inline keyboards/cosmo65/config.h:31]:
```suggestion
```

File: keyboards/cosmo65/config.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* RGB Lighting */
+#define RGB_DI_PIN F7
+#define RGBLED_NUM 16
+#define RGBLIGHT_ANIMATIONS
```
Comment (waffle87) [inline keyboards/cosmo65/config.h:22]:
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
```

File: keyboards/cosmo65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,20 @@
+RGBLIGHT_ENABLE = yes
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/cosmo65/rules.mk:3]:
```suggestion
```

File: keyboards/cosmo65/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/cosmo65/keymaps/via/keymap.c:1]: This file is missing a license header.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/cosmo65/cosmo65.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+GNU General Public License for more details.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see http://www.gnu.org/licenses/.
+*/
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_all( \
```
Comment (drashna) [inline keyboards/cosmo65/cosmo65.h:19]: This, and the cosmo65.c files can be removed, actually.

File: keyboards/cosmo65/info.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    "manufacturer": "BentoBox Studio",
+    "url": "bentoboxstudio.xyz",
+    "maintainer": "BentoBoxStudio",
+    "usb": {
+        "vid": "0x7070",
+        "pid": "0x6331",
+        "device_version": "0.0.1"
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (drashna) [inline keyboards/cosmo65/info.json:12]: if there are no other supported layouts here, this should be `LAYOUT`.  `LAYOUT_all` is generally for keyboards that have multiple layouts (the dz60's are a good example of that)

File: keyboards/cosmo65/cosmo65.h (hunk 5 above)
Comment (BentoBoxStudio) [reply keyboards/cosmo65/cosmo65.h:19]: For the "This," at the beginning of your sentence, are you still referring to changing all the LAYOUT_all to LAYOUT? Thanks

File: keyboards/cosmo65/info.json (hunk 6 above)
Comment (BentoBoxStudio) [reply keyboards/cosmo65/info.json]: I have changed them all, marking as resolved.

**Comment by BentoBoxStudio** [comment]
Are there any more changes I need to make to the files or do they look good?

**Review by drashna — APPROVED** [review]

File: keyboards/cosmo65/readme.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# cosmo65 by BentoBox Studio
+
+![cosmo65](https://imgur.com/Aav54LWh.png)
+
+*A 65% stacked acrylic keyboard with a unique space-themed look.*
+
+* Keyboard Maintainer: [Bento Box Studio](https://github.com/BentoBoxStudio)
+* Hardware Supported: *Supports the cosmo65 keyboard*
```
Comment (fauxpark) [inline keyboards/cosmo65/readme.md:8]:
```suggestion
A 65% stacked acrylic keyboard with a unique space-themed look.

* Keyboard Maintainer: [Bento Box Studio](https://github.com/BentoBoxStudio)
* Hardware Supported: Supports the cosmo65 keyboard
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cosmo65/readme.md (hunk 7 above)
Comment (BentoBoxStudio) [reply keyboards/cosmo65/readme.md]: Ok, it should be changed now.

**Comment by BentoBoxStudio** [comment]
@fauxpark Do you happen to know why the CI Build is failing now that I have committed some changes to the readme? I'm confused about why it can't update the submodules for my most recent commit. Thanks.

File: keyboards/cosmo65/rules.mk [hunk 8]
```diff
@@ -0,0 +1,17 @@
+# MCU Name
+MCU = atmega32u4
+
+# Bootloader
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
```
Comment (fauxpark) [inline keyboards/cosmo65/rules.mk:8]:
```suggestion
#   change yes to no to disable
#
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by fauxpark** [comment]
CI shows as passed for me. I'm guessing someone reran it.

**Pull request merged**